optimum/bettertransformer/transformation.py

```diff
@@ -0,0 +1,190 @@
+    sort_fn = lambda list_modules: [module.__class__.__name__ for module in list_modules]  # noqa: E731
+
+    for key in dict_named_module.keys():
+        if isinstance(dict_named_module[key], torch.nn.ModuleList) and all(
+            "LayerFast" in module_name for module_name in sort_fn(dict_named_module[key])
+        ):
+            setattr(dict_named_module[key][-1], "is_last_layer", True)
+            return True
+    return False
+
+
+@check_if_pytorch_greater_112()
+class BetterTransformer(object):
+    r"""
+    A conversion wrapper that takes as an input the `transformers` model to be converted
+    and returns the converted `Fast` model. The `Fast` model is based on the `BetterTransformer`
+    recently released by PyTorch from its 1.12 version:
+    https://pytorch.org/blog/a-better-transformer-for-fast-transformer-encoder-inference/
+
+    Original PR from: https://github.com/huggingface/transformers/pull/19553 adapted and wrapped in
+    this script.
+    """
+
+    def transform(model, keep_original_model=False, max_memory=None):
+        r"""
+        Conversion script from `transformers` model to its BetterTransformers version
+
+        Args:
+            model, (`torch.nn.Module`, **required**):
+                Original `transformers` model
+            keep_original_model (`bool`, *optional):
+                whether to keep or override the original model - essentially
+                for memory efficiency reasons
+            max_memory (`dict`, *optional*):
+                Same argument as `max_memory` argument from `.from_pretrained` function
+                in `transformers`.
+        """
+
+        # Check if we have to load the model using `accelerate`
+        if hasattr(model, "hf_device_map"):
+            load_accelerate = True
+        else:
+            load_accelerate = False
+
+        if keep_original_model:
+            model_fast = deepcopy(model)
+            model_fast = replace_to_fast(model_fast).eval()
+        else:
+            model_fast = replace_to_fast(model).eval()
+
+        successfully_converted_model = set_last_layer(model_fast)
+        if not successfully_converted_model:
+            raise NotImplementedError(
+                f"The Better Transformers implementation for the model {model.__class__.__name__} has not been"
+                f"implemented yet. Please open an issue requesting the addition of this model with its `Fast`"
+                f"implementation."
+            )
+
+        # Step 6: Add a class arguments, we might need to identify whether the model
+        # has been correctly converted to its `Fast` version.
+        setattr(model_fast, "is_fast", True)
+
+        # Step 7: dispatch model if `accelerate` is enabled
+        if load_accelerate:
+            device_map_bt = infer_auto_device_map(model_fast, max_memory=max_memory)
+
+            remove_hooks_recursive(model_fast)
+
+            model_fast = dispatch_model(model_fast, device_map_bt)
```

Comment:
Ideally you would prefer to do this layer by layer to avoid the blowup in RAM for very large models.